pkg/util/winutil/iisconfig.go

```diff
@@ -0,0 +1,171 @@
+// changes, and reload the configuration when it changes.  It provides additional
+// methods for getting specific configuration items
+type DynamicIISConfig struct {
+	watcher      *fsnotify.Watcher
+	path         string
+	wg           sync.WaitGroup
+	mux          sync.Mutex
+	stopChannel  chan bool
+	xmlcfg       *iisConfiguration
+	siteIDToName map[uint32]string
+}
+
+func NewDynamicIISConfig() (*DynamicIISConfig, error) {
+	iiscfg := &DynamicIISConfig{
+		stopChannel: make(chan bool),
+	}
+	var err error
+
+	iiscfg.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// check for existence
+	_, err = os.Stat(iisCfgPath)
+	if os.IsNotExist(err) {
+		return nil, err
+	} else if err != nil {
+		return nil, err
+	}
+	iiscfg.path = iisCfgPath
+	return iiscfg, nil
+}
+
+func (iiscfg *DynamicIISConfig) Start() error {
+	if iiscfg == nil {
+		return fmt.Errorf("Null config")
+	}
+	// set the filepath
+	err := iiscfg.watcher.Add(iiscfg.path)
+	if err != nil {
+		return err
+	}
+	err = iiscfg.readXmlConfig()
+	if err != nil {
+		return err
+	}
+	iiscfg.wg.Add(1)
+	go func() {
+		defer iiscfg.wg.Done()
+		for {
+			select {
+			case event := <-iiscfg.watcher.Events:
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					iiscfg.readXmlConfig()
+				}
+			case err = <-iiscfg.watcher.Errors:
+				return
+			case <-iiscfg.stopChannel:
+				return
+			}
+		}
+
+	}()
+	return nil
+}
+
+func (iiscfg *DynamicIISConfig) Stop() {
+	iiscfg.stopChannel <- true
+	iiscfg.wg.Wait()
+	return
+}
+
+type iisVirtualDirectory struct {
+	Path         string `xml:"path,attr"`
+	PhysicalPath string `xml:"physicalPath,attr"`
+}
+type iisBinding struct {
+	Protocol    string `xml:"protocol,attr"`
+	BindingInfo string `xml:"bindingInformation,attr"`
+}
+type iisApplication struct {
+	XMLName     xml.Name              `xml:"application"`
+	Path        string                `xml:"path,attr"`
+	AppPool     string                `xml:"applicationPool,attr"`
+	VirtualDirs []iisVirtualDirectory `xml:"virtualDirectory"`
+}
+type iisSite struct {
+	Name        string `xml:"name,attr"`
+	SiteID      string `xml:"id,attr"`
+	Application iisApplication
+	Bindings    []iisBinding `xml:"bindings>binding"`
+}
+type iisSystemApplicationHost struct {
+	XMLName xml.Name  `xml:"system.applicationHost"`
+	Sites   []iisSite `xml:"sites>site"`
+}
+type iisConfiguration struct {
+	XMLName         xml.Name `xml:"configuration"`
+	ApplicationHost iisSystemApplicationHost
+}
+
+func (iiscfg *DynamicIISConfig) readXmlConfig() error {
+	var newcfg iisConfiguration
+	f, err := ioutil.ReadFile(iiscfg.path)
+	if err != nil {
+		return err
+	}
+	err = xml.Unmarshal(f, &newcfg)
+	if err != nil {
+		return err
+	}
+	idmap := make(map[uint32]string)
+
+	for _, site := range newcfg.ApplicationHost.Sites {
+		id, err := strconv.Atoi(site.SiteID)
+		if err != nil {
+			return err
+		}
+		idmap[uint32(id)] = strings.Replace(site.Name, " ", "_", -1)
```

Comment:
why not store the site name as is?

Reply:
because the site name becomes a tag, and tags shouldn't hvae spaces.  And it's much more efficient to do it once when we parse it than to to the string convert every time we access it.

Reply:
Are the `SiteName` or `AppPool` variables used anywhere else that might expect the real value instead of a tag value?

If you must store them as tag values the variable name should reflect that (e.g. `SiteName_tag`)

Reply:
it is only used as a tag.
I can change the variable name.
The reason I don't want to make the transformation when the tag is made is that is potentially done thousands of times per second, vs. doing once when we read.